ColabTool/Shared/PostDetails/index.jsx

```diff
@@ -80,94 +82,93 @@ const renderControls = ({
       <Text size={'small'}> Deleting... </Text>
     );
   }
+
+  if (manager && isWorking && !isDeleting) {
+    return (
+      <Text size={'small'}> Approving... </Text>
+    );
+  }
+
   return (
     <div>
-      {renderEdit({
-        onEditClick,
-      })}
       {renderDelete({
         isConfirmingDelete,
         onCancelConfirmClick,
         onDeleteConfirmClick,
         onDeleteClick,
       })}
+      {renderEdit({
+        onEditClick,
+      })}
+      {manager && renderApproval({
+        onApproveClick,
+      })}
     </div>
   );
 };
 
-const renderPostTypeIcon = (postType) => {
-  switch (postType) {
-    case 'image':
-      return (<ImageIcon size={'small'} />);
-    case 'retweet':
-      return (<RetweetIcon size={'small'} />);
-    default:
-      return (<LinkIcon size={'small'} />);
-  }
-};
-
 /* eslint-enable react/prop-types */
 
 const PostDetails = ({
   isDeleting,
   isConfirmingDelete,
+  isWorking,
+  manager,
+  onApproveClick,
   onCancelConfirmClick,
   onDeleteClick,
   onDeleteConfirmClick,
   onEditClick,
-  profile,
-  postType,
+  draftDetails,
 }) =>
   <div className={style['post-details']}>
+    <div className={style['post-action-details']}>
```

Comment:
Heya @stevemdixon - did a bit of testing with this one and removed it and it looks like we do need it so I'll go ahead and keep that one 😄